### PR TITLE
New Metrics Query protocol 

### PR DESCRIPTION
**Pull request by wu-sheng**
This protocol provides the capability, client side could use name rather than ID to query metrics. This is much easier and suitable for the new metrics system.

### CODE REVIEW

**Comment by Fine0830** [comment]
From the perspective of UI, LGTM.

File: metrics-v2.graphqls [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    # Read metrics single value in the duration of required metrics for
+    readMetricsValue(metrics: MetricsCondition!, duration: Duration!): Int!
+    readMetricsValues(metrics: MetricsCondition!, duration: Duration!): [MetricsValues!]!
+    sortMetrics(metrics: MetricsCondition!, duration: Duration!): [TopNRecords!]!
+    # Read value in the given time duration, usually as a linear.
+    # labels: the labels you need to query.
+    readLabeledMetricsValues(metrics: MetricsCondition!, labels: [String!]!, duration: Duration!): [MetricsValues!]!
+    # Heatmap is bucket based value statistic result.
+    readHeatMap(metrics: MetricsCondition!, duration: Duration!): HeatMap
```
Comment (wu-sheng) [inline metrics-v2.graphqls:125]: @lunchboxav The heatmap you asked.

File: aggregation.graphqls [hunk 2]
```diff
@@ -14,6 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Legacy metrics query protocol
+# Replyced by the metrics-v2 in the future
```
Comment (kezhenxu94) [inline aggregation.graphqls:18]: Typo

File: metrics-v2.graphqls [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    id: ID!
+    # Usually an integer as this is metrics.
+    value: String
+    # Have value, Only if the record has related trace id.
+    # UI should show this as an attached value.
+    refId: ID
+}
+
+extend type Query {
+    # Metrics definition metadata query. Respont the metrics type which determines the suitable query methods.
```
Comment (kezhenxu94) [inline metrics-v2.graphqls:114]: typo `Respont`

**Comment by wu-sheng** [comment]
@kezhenxu94 Fixed all typos you pointed out.

**Pull request merged**